tree/ntuple/v7/inc/ROOT/RFieldBase.hxx

```diff
@@ -244,6 +245,13 @@ private:
       }
    }
 
+   void SetEmulatedFromType(std::string_view type)
+   {
+      assert(fType.empty());
+      fTraits |= kTraitEmulatedField;
+      fType = type;
+   }
+
```

Comment:
Do we need this generically in `RFieldBase` or can this move to `RRecordField`? potentially as a constructor option or with a factory method (was we have `RVectorField::CreateUntyped()`)